ga/client/bitrateadaptation.cpp

```diff
@@ -0,0 +1,203 @@
+/*
+ * Copyright (c) 2016 Chun-Ying Huang
+ *
+ * This file is part of GamingAnywhere (GA).
+ *
+ * GA is free software; you can redistribute it and/or modify it
+ * under the terms of the 3-clause BSD License as published by the
+ * Free Software Foundation: http://directory.fsf.org/wiki/License:BSD_3Clause
+ *
+ * GA is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ *
+ * You should have received a copy of the 3-clause BSD License along with GA;
+ * if not, write to the Free Software Foundation, Inc.,
+ * 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+ */
+
+#include <stdlib.h>
+#ifndef WIN32
+#include <unistd.h>
+#endif
+
+#include "ga-common.h"
+#include "controller.h"
+#include "rttserver.h"
+#include "bitrateadaptation.h"
+
+typedef struct bbr_btlbw_record_s {
+	struct timeval rcvtime;
+	unsigned int pktsize;
+	unsigned int timeelapsed; // Time in usec
+	unsigned int deliveryrate; // In bytes per sec
+}	bbr_record_t;
+
+static struct bbr_btlbw_record_s bbr_btlbw[BBR_BTLBW_MAX];
+static unsigned int latest_throughput; // Not thread safe.
+
+void
+bbr_update(unsigned int ssrc, unsigned int seq, struct timeval rcvtv, unsigned int timestamp, unsigned int pktsize) {
+	// assume ssrc is always video source.
+	static unsigned int bbr_btlbw_start = 0;
+	static unsigned int bbr_btlbw_head = 0;
+	static unsigned int last_pkt_timestamp = 0;
```

Comment:
Even though they now have the static keyword, they are initialized on every call of this function effectively making them not static.